espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.java

```diff
@@ -127,6 +134,46 @@ private void buildXML(Document doc, Element parentElement, View view) {
             parentElement.appendChild(element);
         }
 
+        // If it's an AdapterView, get the adapters as a String
+        try {
+            AdapterView adapterView = (AdapterView) view;
+            Adapter adapter = adapterView.getAdapter();
+            StringBuilder adapterData = new StringBuilder();
+            for (int i = 0; i < adapter.getCount(); i++) {
+                Object adapterItem = adapter.getItem(i);
+                adapterData.append(adapterItem);
+                if (i < adapter.getCount() - 1) {
+                    adapterData.append(",");
+                }
+
+                // Get the type of the adapter item
+                if (i == 0) {
```

Comment:
`adapterItemType = adapterItem.getClass().getShortName()`
Also, it might be handy to apply the same class name filter that we apply for other classes